src/Nethermind/Nethermind.Specs/GoerliSpecProvider.cs

```diff
@@ -36,27 +36,35 @@ public void UpdateMergeTransitionInfo(long? blockNumber, UInt256? terminalTotalD
 
         private IReleaseSpec LondonNoBomb { get; } = London.Instance;
 
-        public IReleaseSpec GetSpec(ForkActivation forkActivation) =>
-            forkActivation.BlockNumber switch
+
+        public IReleaseSpec GetSpec(ForkActivation forkActivation)
+        {
+            if (forkActivation.Timestamp >= ShanghaiTimestamp)
+                return Shanghai.Instance;
+
+            return forkActivation.BlockNumber switch
             {
                 < IstanbulBlockNumber => GenesisSpec,
                 < BerlinBlockNumber => IstanbulNoBomb,
-                < LondonBlockNumber => BerlinNoBomb,
-                _ => LondonNoBomb
+                < LondonBlockNumber => Berlin.Instance,
+                _ => London.Instance
```

Comment:
shouldn't we have something for Shanghai here?

Reply:
It's there, a bit above.

Reply:
Refactored